### PR TITLE
fix: cleanup getMaxContainerHeight and getMaxContainerWidth

### PR DESCRIPTION
**Pull request by ad1992**
- [x] cleanup getMaxContainerHeight and getMaxContainerWidth
- [x] rename getMaxContainerWidth -> getBoundTextMaxWidth and getMaxContainerWidth-> getBoundTextMaxHeight
- [x] Add missing spec

### CODE REVIEW

File: src/element/textElement.ts [hunk 1]
```diff
@@ -890,18 +899,10 @@ export const computeContainerDimensionForBoundText = (
...
   if (isArrowElement(container)) {
-    const containerWidth = width - BOUND_TEXT_PADDING * 8 * 2;
-    if (containerWidth <= 0) {
-      const boundText = getBoundTextElement(container);
-      if (boundText) {
-        return boundText.width;
-      }
-      return BOUND_TEXT_PADDING * 8 * 2;
-    }
-    return containerWidth;
```
Comment (dwelle) [inline src/element/textElement.ts:904]: why was this needed and is not needed anymore?

Comment (ad1992) [reply src/element/textElement.ts:904]: This was never needed actually coz if you notice, `getContainerDims` will never return dimensions <=0 for arrow the min width returned is `300` so this code never gets executed.
So I believe only when that check for 300 wasn't there only then this was being used otherwise not.

**Review by dwelle — APPROVED** [review]
